### PR TITLE
*-DbaDbOrphanUser - Improve the workflow and fix some bugs

### LINKED ISSUE #6459: [Bug] Repair-DbaDbOrphanUser ignores -RemoveNotExisting switch
### Environmental information

```
#### PowerShell version:

Name                           Value
----                           -----
PSVersion                      5.1.18362.628
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.628
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


#### dbatools Module version:

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.101\dbatools.psd1
Version : 1.0.101

```

#### SQL Server:

```sql
Microsoft SQL Server 2014 (SP3-GDR) (KB4532095) - 12.0.6118.4 (X64)   Dec 12 2019 21:46:15   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor) 
```

#### Host used

- [ X ] powershell.exe
- [ ] ISE
- [ ] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [ ] Still buggy in powershell.exe

#### Errors Received

No errors received.

#### Steps to Reproduce

Repair-DbaDbOrphanUser -SqlInstance <instance> -Database <database> -RemoveNotExisting

#### Expected Behavior

When executed with -WhatIf, output looks to be what was expected.

What if: Performing the operation "Mapping user 'user'" on target "database".
What if: Performing the operation "Remove-DbaDbOrphanUser" on target "database".

#### Actual Behavior

Orphaned user mapped successfully.

Used the following:
Get-DbaDbOrphanUser -SqlInstance <instance> -Database <database>

Returns items I expected to have been removed.

-EnableException comes back clear.

Manually executed:
Remove-DbaDbOrphanUser -SqlInstance <instance> -Database <database>

Works as I would have expected.

### PR DESCRIPTION
**Pull request by andreasjordan**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6459 )

The original bug in the linked issue was just the missung `$users = $User` because the names of these parameters are not the same in all commands.

While testing I found that `Get-DbaDbOrphanUser` only lists the orphan users from windows logins so I changed that code. Does anyone know why there was a `-eq $false`?

And then I changed Repair- and Remove- to use Get- to reduce duplicate code. Therefor Get- now returns the SMO of the user as well (Like Get-DbaDbRoleMember does).

I don't know if this commands handles all cases correctly. I found this link that might help: https://www.mssqltips.com/sqlservertip/5221/different-ways-to-find-sql-server-orphaned-users/

I always tell my students and clients how to avoid orphan users so I don't have any interest in these commands. But if someone has the time and the interest - there is room for improvement in these commands.

### CODE REVIEW

**Comment by andreasjordan** [comment]
Maybe look at #7253 as well...

**Comment by andreasjordan** [comment]
Ok, all tests for Remove-DbaDbOrphanUser fail - just have no time to work on that today or tomorrow. Will continue next week...

**Comment by potatoqualitee** [comment]
The code LGTM. If anyone else approves, I'm happy to merge. I believe @ClaudioESSilva did a lot of orphan work.

File: functions/Get-DbaDbOrphanUser.ps1 [hunk 1]
```diff
@@ -91,7 +91,7 @@ function Get-DbaDbOrphanUser {
                     try {
                         Write-Message -Level Verbose -Message "Validating users on database '$db'."
                         $UsersToWork = @()
-                        $UsersToWork += $db.Users | Where-Object { $_.Login -eq "" -and ($_.ID -gt 4) -and (($_.Sid.Length -gt 16 -and $_.LoginType -in @([Microsoft.SqlServer.Management.Smo.LoginType]::SqlLogin, [Microsoft.SqlServer.Management.Smo.LoginType]::Certificate)) -eq $false) }
+                        $UsersToWork += $db.Users | Where-Object { ($_.Login -eq "") -and ($_.ID -gt 4) -and ($_.Sid.Length -eq 16) -and ($_.LoginType -in @([Microsoft.SqlServer.Management.Smo.LoginType]::SqlLogin, [Microsoft.SqlServer.Management.Smo.LoginType]::Certificate)) }
```
Comment (wsmelton) [inline functions/Get-DbaDbOrphanUser.ps1:94]:
```
$_.LoginType -in @([Microsoft.SqlServer.Management.Smo.LoginType]::SqlLogin, [Microsoft.SqlServer.Management.Smo.LoginType]::Certificate))
```

Is the same as doing:

```
$_.LoginType -in 'SqlLogin','Certificate'
```

The later is much easier to read and understand for users, in general.

File: functions/Get-DbaDbOrphanUser.ps1 [hunk 2]
```diff
@@ -102,7 +102,8 @@ function Get-DbaDbOrphanUser {
                                     SqlInstance  = $server.DomainInstanceName
                                     DatabaseName = $db.Name
                                     User         = $user.Name
-                                }
+                                    SMOUser      = $user
```
Comment (wsmelton) [inline functions/Get-DbaDbOrphanUser.ps1:105]:
```suggestion
                                    SmoUser      = $user
```

File: functions/Get-DbaDbOrphanUser.ps1 (hunk 1 above)
Comment (andreasjordan) [reply functions/Get-DbaDbOrphanUser.ps1:94]: Yes, will change that.

File: functions/Get-DbaDbOrphanUser.ps1 (hunk 2 above)
Comment (andreasjordan) [reply functions/Get-DbaDbOrphanUser.ps1:105]: @potatoqualitee What do you like? The *RoleMember commands use "SMOUser" so I used it here. "SmoUser" is ok for me as well - but maybe then change it everywhere?

Comment (potatoqualitee) [reply functions/Get-DbaDbOrphanUser.ps1:105]: Yes, today I like SmoUser too 😅

Comment (andreasjordan) [reply functions/Get-DbaDbOrphanUser.ps1:105]: Should I change the name in *RoleMember commands also? Within this PR?

Comment (potatoqualitee) [reply functions/Get-DbaDbOrphanUser.ps1]: You can if you'd like, yes. Once tests pass, I'll merge them all.

Comment (andreasjordan) [reply functions/Get-DbaDbOrphanUser.ps1]: Ok, done.

**Comment by potatoqualitee** [comment]
sweet, thanks so much!

**Pull request merged**